### PR TITLE
Simplify commands used in README instructions

### PR DESCRIPTION
**Pull request by michaelfig**
Lightly tested, but this is the approach I'd like to take.

@sufyaankhan, please take a look at README.md for how this architecture simplifies the flow.

Recommendations appreciated.

### CODE REVIEW

**Comment by michaelfig** [comment]
Here is how it currently fails (because `@agoric/xsnap` is broken in the `beta` packages):

```console
$ yarn start
yarn run v1.22.5
$ concurrently --kill-others -c "bgBlue.bold,bgMagenta,bgYellow.bold" --names "SIM,DEP,UI " "yarn start:sim" "yarn start:deploy" "yarn start:ui"
$ agoric deploy contract/deploy.js api/deploy.js
$ agoric start --reset
$ cd ui && yarn start
$ parcel serve --host=127.0.0.1 --port=3000 public/index.html
[DEP] Open CapTP connection to ws://127.0.0.1:8000/private/captp...agoric: start: removing _agstate/agoric-servers/dev
[SIM] agoric: start: rm -rf _agstate/agoric-servers/dev
[SIM] agoric: start: initializing dev
[SIM] agoric: start: /Users/michael/agoric/dapp-fungible-faucet/node_modules/@agoric/solo/src/entrypoint.js init dev --egresses=fake --webport=8000
[UI ] Server running at http://127.0.0.1:3000
[UI ] ✨ Built in 7ms
[SIM] agoric: start: setting sim chain with 0 second delay
[SIM] agoric: start: /Users/michael/agoric/dapp-fungible-faucet/node_modules/@agoric/solo/src/entrypoint.js set-fake-chain --delay=0 sim-chain
[DEP] .agoric: start: /Users/michael/agoric/dapp-fungible-faucet/node_modules/@agoric/solo/src/entrypoint.js start
[DEP] ......Loading makeSlogSender from @agoric/telemetry/src/flight-recorder.js
[SIM] (Error#1)
[SIM] Error#1: spawn /Users/michael/agoric/dapp-fungible-faucet/node_modules/@agoric/xsnap/xsnap-native/xsnap/build/bin/mac/release/xsnap-worker ENOENT
[SIM] 
error Command failed with exit code 255.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[SIM] yarn start:sim exited with code 255
--> Sending SIGTERM to other processes..
[DEP] yarn start:deploy exited with code 1
--> Sending SIGTERM to other processes..
[UI ] yarn start:ui exited with code 1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
snow:dapp-fungible-faucet michael$ 
```

**Comment by michaelfig** [comment]
After `yarn agoric install agoric-upgrade-12` didn't fix the issue, I found that the actual problem was in the `@agoric/xsnap/package.json`.  It was using `git status` to detect if the package was in development vs. production mode, and that gave wrong answers when installed within a dapp project.  I switched it to use `test -d ./test` to detect development mode (since the `./test` directory is omitted from the NPM package).

Now I was able to run the README instructions to completion (contract successfully deployed), and the next broken thing appeared to be a problem with running an older `ses` in the dapp UI.

I call this a victory.

**Review by mhofman — DISMISSED** [review]
Use `getting-started-dev` instead of `beta`

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -13,50 +13,57 @@ jobs:
...
         echo "/home/runner/bin" >> $GITHUB_PATH
       working-directory: ./agoric-sdk
 
-    - name: agoric install
-      run: agoric install
+    - name: install packages
+      run: |
+        if ${{ steps.get-branch.outputs.result == 'NONE' }}; then
+          yarn install
+          echo "AGORIC_CMD=[\"$PWD/node_modules/.bin/agoric\"]" >> $GITHUB_ENV
```
Comment (mhofman) [inline .github/workflows/integration-test.yml:63]: Is there a way we can avoid this workaround? What is it needed for anyway?

File: api/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "dapp-fungible-faucet-api",
+  "name": "@agoric/dapp-fungible-faucet-api",
```
Comment (mhofman) [inline api/package.json:2]: I kinda wish we used another scope for these example dapps

File: api/package.json [hunk 3]
```diff
@@ -13,26 +13,12 @@
...
-    "eslint-config-prettier": "^6.15.0",
-    "eslint-plugin-import": "^2.26.0",
-    "eslint-plugin-prettier": "^4.0.0",
-    "eslint-plugin-jsdoc": "^39.3.3",
-    "prettier": "^2.5.1"
-  },
   "dependencies": {
     "@endo/eventual-send": "^0.15.5",
     "@endo/marshal": "^0.6.9",
+    "@agoric/dapp-fungible-faucet-ui": "^0.0.1",
```
Comment (mhofman) [inline api/package.json]: The API having a dependency on the UI seems backwards

File: package.json [hunk 4]
```diff
@@ -55,7 +57,8 @@
   "eslintConfig": {
     "extends": [
       "@agoric"
-    ]
+    ],
+    "parser": "@typescript-eslint/parser"
```
Comment (mhofman) [inline package.json:61]: 👍

File: api/deploy.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 
+/* global process */
 // Agoric Dapp api deployment script
 
 import fs from 'fs';
 import { E } from '@endo/eventual-send';
 import '@agoric/zoe/exported.js';
 
-import installationConstants from '../ui/public/conf/installationConstants.js';
+import installationConstants from "@agoric/dapp-fungible-faucet-ui/public/conf/installationConstants.js";
```
Comment (mhofman) [inline api/deploy.js]: Oh that's the dependency. Yeah this really feels backwards

**Review by mhofman — COMMENTED** [review]
Quick review. Looks great so far. I have to dig out my notes of when I went through the same a couple month ago.

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (michaelfig) [reply .github/workflows/integration-test.yml:63]: The agoric-cli/tools integration test needs to know how to find the CLI, and it defaults to using the PATH.  I didn't want to munge with the `$GITHUB_PATH` file, so I used the same `$AGORIC_CMD` environment variable that the `agoric-sdk/scripts/registry.sh` uses.

File: api/package.json (hunk 2 above)
Comment (michaelfig) [reply api/package.json:2]: Why?  The names get replaced when people do `npm create` or `agoric init`.

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (mhofman) [reply .github/workflows/integration-test.yml:63]: Oh I didn't see this used anywhere, so I thought it was a magic env

File: api/package.json (hunk 3 above)
Comment (michaelfig) [reply api/package.json]: I can try to flip this around, but it was this way when I arrived here.

File: api/package.json (hunk 2 above)
Comment (mhofman) [reply api/package.json:2]: I guess if these packages are never published to NPM, that's fine.

File: api/package.json (hunk 3 above)
Comment (mhofman) [reply api/package.json]: Yeah it's out of scope, but maybe we should have an issue to track

**Comment by mhofman** [comment]
For reference, I had done something similar to dapp-card-store a couple month ago, but our approach on a some deps seem to have been different in a couple places: https://github.com/Agoric/dapp-card-store/compare/main...mhofman/npm-latest

**Comment by mhofman** [comment]
@dckc another set of dapp work for NPM compatibility during the hackathon

**Comment by dckc** [comment]
I'd be inclined to mothball (archive) this dapp. I don't think it's worth maintaining.

**Comment by michaelfig** [comment]
I'm closing this one since we're not putting more energy into dapp-fungible-faucet.  The history here might be useful for someone in the future, though.

**Pull request closed without merging**